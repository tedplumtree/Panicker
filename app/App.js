import React from 'react';
import { ThemeProvider } from 'react-native-material-ui';
import { Provider as StoreProvider } from 'react-redux';
import { createStore } from 'redux';
import { MaterialDesignTheme } from './config/styles';
import SettingsAndActivateScreen from './screens/SettingsAndActivateScreen';
import combinedReducers from './reducers/index';

const store = createStore(combinedReducers);

const App = () => (
  <StoreProvider store={store}>
    <ThemeProvider uiTheme={MaterialDesignTheme}>
      <SettingsAndActivateScreen />
    </ThemeProvider>
  </StoreProvider>
);

export default App;
