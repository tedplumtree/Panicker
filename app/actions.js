export const TOGGLE_SENSOR = 'TOGGLE_SENSOR';

export const toggleSensor = () => ({
  type: TOGGLE_SENSOR,
});
