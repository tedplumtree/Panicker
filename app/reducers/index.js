import { combineReducers } from 'redux';
import sensor from './sensor';

const combined = combineReducers({
  sensor,
});

export default combined;
