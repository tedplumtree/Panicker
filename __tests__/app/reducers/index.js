import { toggleSensor } from '../../../app/actions';
import combinedReducers from '../../../app/reducers/index';

const combinedState = {
  sensor: {
    active: false,
  },
};

describe('sensor', () => {
  const generateInitialState = sensorState => ({
    ...combinedState,
    sensor: sensorState,
  });

  describe('when toggled', () => {
    describe('when sensor is inactive', () => {
      const initialState = generateInitialState({ active: false });

      it('correctly reduces the action', () => {
        expect(
          combinedReducers(initialState, toggleSensor()),
        ).toMatchSnapshot();
      });
    });

    describe('when sensor is active', () => {
      const initialState = generateInitialState({ active: true });

      it('correctly reduces the action', () => {
        expect(
          combinedReducers(initialState, toggleSensor()),
        ).toMatchSnapshot();
      });
    });
  });

  describe('when passed a nonexistent action', () => {
    const action = { type: 'NONEXISTENT_ACTION' };

    describe('when sensor is in initial state', () => {
      const initialState = generateInitialState(undefined);

      // Since in Redux, it makes most sense for the default reduction to
      // yield the initial state, this is a reasonable pattern for actually
      // testing what the initial state is
      it("doesn't do anything", () => {
        expect(
          combinedReducers(initialState, action),
        ).toMatchSnapshot();
      });
    });

    describe('when sensor is inactive', () => {
      const initialState = generateInitialState({ active: false });

      it("doesn't do anything", () => {
        expect(
          combinedReducers(initialState, action),
        ).toMatchSnapshot();
      });
    });

    describe('when sensor is active', () => {
      const initialState = generateInitialState({ active: true });

      it("doesn't do anything", () => {
        expect(
          combinedReducers(initialState, action),
        ).toMatchSnapshot();
      });
    });
  });
});
