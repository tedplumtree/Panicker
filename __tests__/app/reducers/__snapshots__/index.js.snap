// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sensor when passed a nonexistent action when sensor is active doesn't do anything 1`] = `
Object {
  "sensor": Object {
    "active": true,
  },
}
`;

exports[`sensor when passed a nonexistent action when sensor is in initial state doesn't do anything 1`] = `
Object {
  "sensor": Object {
    "active": false,
  },
}
`;

exports[`sensor when passed a nonexistent action when sensor is inactive doesn't do anything 1`] = `
Object {
  "sensor": Object {
    "active": false,
  },
}
`;

exports[`sensor when toggled when sensor is active correctly reduces the action 1`] = `
Object {
  "sensor": Object {
    "active": false,
  },
}
`;

exports[`sensor when toggled when sensor is inactive correctly reduces the action 1`] = `
Object {
  "sensor": Object {
    "active": true,
  },
}
`;
